### ISSUE TITLE
gradient accumulation

### ISSUE THREAD

**Issue by wanghaijie2017** [Navhkrin/Bts-PyTorch#2]
I noticed that your code uses gradient accumulation instead of batch size=4. Although this method can increase the batch size, it will still maintain the original size when processing the BN layer. It may cause performance degradation.

**Comment by Navhkrin** [comment]
I added gradient accumulation is to artificially increase batch size on a single GPU with low VRAM. Pytorch batch norm uses running mean and variance, exponentially calculated from multiple passes, so it is a bit more stable but yes, this will not be as stable as using a bigger batch size without accumulation.
To disable gradient accumulation, set same values to batch_size and effective_batch_size parameters on top of Train.py file.

Im going to add Multi Gpu and Nvidia Apex support on my next commit, but if you need multi gpu support early you can simply change folllowing line on BTS.py -> BtsController
`self.bts = bts_eren().float().cuda()`
to
`self.bts = torch.nn.DataParallel(bts_eren().float().cuda())`
But do note that if you do this, it wont be compatible with pretrained model
